test/core.jl

```diff
@@ -5089,3 +5089,23 @@ m22929_2.x = m22929_1
 @test !isdefined_22929_x(m22929_1)
 @test isdefined_22929_1(m22929_2)
 @test isdefined_22929_x(m22929_2)
+
+# issue 18933
+module GlobalDef18933
+    using Base.Test
+    f() = (global sin; nothing)
```

Comment:
It's possible we'll require explicit qualification for extension anyway, so this might be moot. However, in the meantime it's possible that `global x` should not overwrite "hard-imported" bindings. That will be less breaking, since currently it gives a warning.